### DISCUSSION TITLE
Ldap Extension for /shivammathur/php@7.4

### DISCUSSION THREAD

**Discussion by armandgabriel** [shivammathur/homebrew-extensions#2817] [Q&A] [answered]
Hello there,

I come here after countless hours of searching a fix for adding the 'extension=ldap' for the php env on my mac m1 laptop. I found that the most stable version of php is the one provided by you @shivammathur , installed via brew and managed to start my Laravel application. The next thing is to activate the ldap extension I use for my project. 
After modifying the php.ini file and restarting the service I get this error and I don't know what to do: 
hp --ini

Warning: PHP Startup: Unable to load dynamic library 'ldap' (tried: /opt/homebrew/lib/php/pecl/20190902/ldap (dlopen(/opt/homebrew/lib/php/pecl/20190902/ldap, 0x0009): tried: '/opt/homebrew/lib/php/pecl/20190902/ldap' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/lib/php/pecl/20190902/ldap' (no such file), '/opt/homebrew/lib/php/pecl/20190902/ldap' (no such file)), /opt/homebrew/lib/php/pecl/20190902/ldap.so (dlopen(/opt/homebrew/lib/php/pecl/20190902/ldap.so, 0x0009): tried: '/opt/homebrew/lib/php/pecl/20190902/ldap.so' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/opt/homebrew/lib/php/pecl/20190902/ldap.so' (no such file), '/opt/homebrew/lib/php/pecl/20190902/ldap.so' (no such file))) in Unknown on line 0
Configuration File (php.ini) Path: /opt/homebrew/etc/php/7.4
Loaded Configuration File:         /opt/homebrew/etc/php/7.4/php.ini
Scan for additional .ini files in: /opt/homebrew/etc/php/7.4/conf.d
Additional .ini files parsed:      /opt/homebrew/etc/php/7.4/conf.d/error_log.ini,
/opt/homebrew/etc/php/7.4/conf.d/ext-opcache.ini,
/opt/homebrew/etc/php/7.4/conf.d/php-memory-limits.ini

Please if anyone has any solutions would help me.

Thank you in advance,
Kind regards,
Armand Camner

**Comment by shivammathur** [comment, marked as answer]
@armandgabriel 
You don't need to add `extension=ldap` in your php.ini. Please remove that.

Ldap extension in these php builds is compiled statically along with PHP. So it should be available once PHP is installed
https://github.com/shivammathur/homebrew-php/blob/master/Formula/php@7.4.rb#L164

Please run `php -m` and ldap should be there.

**Reply by armandgabriel** [reply]
Hello @shivammathur 

Thank you for the swift response!
